pytensor/tensor/extra_ops.py

```diff
@@ -1583,27 +1595,294 @@ def broadcast_shape_iter(
     return tuple(result_dims)
 
 
-def geomspace(start, end, steps, base=10.0):
-    from pytensor.tensor.math import log
+def _check_deprecated_inputs(stop, end, num, steps):
+    if end is not None:
+        warnings.warn(
+            "The 'end' parameter is deprecated and will be removed in a future version. Use 'stop' instead.",
+            DeprecationWarning,
+        )
+        stop = end
+    if steps is not None:
+        warnings.warn(
+            "The 'steps' parameter is deprecated and will be removed in a future version. Use 'num' instead.",
+            DeprecationWarning,
+        )
+        num = steps
+
+    return stop, num
+
+
+def _linspace_core(
+    start: TensorVariable,
+    stop: TensorVariable,
+    num: int,
+    dtype: str,
+    endpoint=True,
+    retstep=False,
+    axis=0,
+) -> TensorVariable | tuple[TensorVariable, TensorVariable]:
+    div = (num - 1) if endpoint else num
+    delta = (stop - start).astype(dtype)
+    samples = ptb.arange(0, num, dtype=dtype).reshape((-1,) + (1,) * delta.ndim)
+
+    step = switch(gt(div, 0), delta / div, np.nan)
+    samples = switch(gt(div, 0), samples * delta / div + start, samples * delta + start)
+    samples = switch(
+        bitwise_and(gt(num, 1), np.asarray(endpoint)),
+        set_subtensor(samples[-1, ...], stop),
+        samples,
+    )
```

Comment:
Put this in a python `if endpoint` statement, not to complicate the graph when not needed? I assume we don't need endpoint to be symbolic since you used `asarray` (I know it can't be a python True)